### PR TITLE
Add macOS to Github Action

### LINKED ISSUE #70: CI needs to test on macOS as well as ubuntu
CI needs to test on macOS as well as ubuntu. This is related to #68

### PR DESCRIPTION
**Pull request by kgerheiser**
Fixes #70

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -3,7 +3,10 @@ on: [push, pull_request]
 
 jobs:
   build:
-    runs-on: ubuntu-20.04
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [macos-10.15,  ubuntu-20.04]
```
Comment (edwardhartnett) [inline .github/workflows/main.yml:9]: So easy! What other OS flavors are available?

**Review by edwardhartnett — APPROVED** [review]

File: .github/workflows/main.yml (hunk 1 above)
Comment (kgerheiser) [reply .github/workflows/main.yml:9]: Ubuntu 16, 18, 20

macOS 10.15, 11.0

Windows Server

Comment (edwardhartnett) [reply .github/workflows/main.yml:9]: Seems like we should use all of them, except Windows.

**Comment by aerorahul** [comment]
can we test with `clang` too please on a macOS?

**Comment by aerorahul** [comment]
The DA applications, believe it or not, tests with `clang`

**Comment by aerorahul** [comment]
This is not right.
there is no sudo apt-get on a macOS.

**Comment by aerorahul** [comment]
Ok. Well you don't have that issue.  I will need to create a separate workflow run for macOS, because we are going to need sudo apt-get and brew for the python PR.

**Comment by aerorahul** [comment]
I was able to run the CI for the python PR.  All tests are passing for both macos and ubuntu. phew!

**Pull request merged**